install_config/advanced_ldap_configuration/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,425 @@
+= Setting up SSSD for LDAP Failover
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+{product-title} provides an
+link:../configuring_authentication.html#LDAPPasswordIdentityProvider[authentication
+provider] for use with Lightweight Directory Access Protocol (LDAP) setups, but
+it can only connect to a single LDAP server. This can be problematic if that
+LDAP server becomes unavailable. System Security Services Daemon (SSSD) can be
+used to solve the issue.
+
+Originally designed to manage local and remote authentication to the host
+operating system, SSSD can now be configured to provide identity,
+authentication, and authorization services to web services like {product-title}.
+SSSD provides advantages over the built-in LDAP provider, including the ability
+to connect to any number of failover LDAP servers, as well as the ability to
+cache authentication attempts in case it can no longer reach any of those
+servers.
+
+The setup for this configuration is advanced and requires a separate
+authentication server (also called an *authenticating proxy*) for
+{product-title} to communicate with. This topic describes how to do this setup
+on a dedicated physical or virtual machine (VM), but the concepts are also
+applicable to a setup in a container.
+
+[[sssd-prerequisites-for-authenticating-proxy-setup]]
+== Prerequisites for Authenticating Proxy Setup
+
+. Before starting setup, you need to know the following information about your
+LDAP server.
++
+* Whether the directory server is powered by
+http://www.freeipa.org/page/Main_Page[FreeIPA], Active Directory, or another
+LDAP solution.
++
+* The Uniform Resource Identifier (URI) for the LDAP server (for example,
+ldap.example.com).
++
+* The location of the CA certificate for the LDAP server.
++
+* Whether the LDAP server corresponds to RFC 2307 or RFC2307bis for user groups.
++
+. Prepare the VMs:
++
+* *_proxy.example.com_*: A VM to use as the authenticating proxy. This machine must
+have at least SSSD 1.12.0 available, which means a fairly recent operating
+system. This topic uses a Red Hat Enterprise Linux 7.2 server for its examples.
++
+* *_openshift.example.com_*: A VM to use to run {product-title}.amp
+
+[NOTE]
+====
+These VMs can be configured to run on the same system, but for the examples used
+in this topic they are kept separate.
+====
+
+[[sssd-phase-1-certificate-generation]]
+== Phase 1: Certificate Generation
+
+. To ensure that communication between the authenticating proxy and
+{product-title} is trustworthy, create a set of Transport Layer Security (TLS)
+certificates to use during the other phases of this setup. We will start by
+using the auto-generated certificates created as part of running:
++
+====
+----
+# openshift start \
+    --public-master=https://openshift.example.com:8443 \
+    --write-config=/etc/origin/
+----
+====
```

Comment:
Probably useful to note explicitly that these steps are being run on the OpenShift system.